cmd/kubeadm/app/cmd/init.go

```diff
@@ -54,28 +55,28 @@ func NewCmdInit(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 
 	cmd.PersistentFlags().StringVar(
 		&s.Secrets.GivenToken, "token", "",
-		`(optional) Shared secret used to secure bootstrap. Will be generated and displayed if not provided.`,
+		`(optional) Shared secret used to secure cluster bootstrap. If none is provided, one will be generated for you.`,
 	)
 	cmd.PersistentFlags().StringSliceVar(
-		advertiseAddrs, "api-advertise-addr", []string{},
-		`(optional) IP address to advertise, in case autodetection fails.`,
+		advertiseAddrs, "api-advertise-addresses", []string{},
+		`(optional) The IP addresses to advertise, in case autodetection fails.`,
 	)
 	cmd.PersistentFlags().StringSliceVar(
-		&s.InitFlags.API.ExternalDNSName, "api-external-dns-name", []string{},
-		`(optional) DNS name to advertise, in case you have configured one yourself.`,
+		&s.InitFlags.API.ExternalDNSNames, "api-external-dns-names", []string{},
+		`(optional) The DNS names to advertise, in case you have configured them yourself.`,
 	)
 
 	cmd.PersistentFlags().IPNetVar(
 		&s.InitFlags.Services.CIDR, "service-cidr", *kubeadmapi.DefaultServicesCIDR,
-		`(optional) use alterantive range of IP address for service VIPs, e.g. "10.16.0.0/12"`,
+		`(optional) use alternative range of IP address for service VIPs, e.g. "10.16.0.0/12"`,
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.Services.DNSDomain, "service-dns-domain", kubeadmapi.DefaultServiceDNSDomain,
-		`(optional) use alterantive domain name for services, e.g. "myorg.internal"`,
+		`(optional) use alternative domain for services, e.g. "myorg.internal"`,
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.CloudProvider, "cloud-provider", "",
-		`(optional) enable cloud proiver features (external load-balancers, storage, etc)`,
+		`(optional) enable a specific cloud provider features (external load-balancers, storage, etc), e.g. "gce"`,
```

Comment:
Again, would be nice if this linked to a doc. I don't actually know how this bit works atm though so I'm struggling to document it :)